lex/lex_utils.ml

```diff
@@ -1,40 +1,62 @@
-open Geneweb
+let () =
 
+(* script adapted for gwrepl *)
+(* 
+  use: cat <script.ml> | [ GWREPL_VERBOSE=1 ] [ GWREPL_FORCE_UNPACK=1 ]
+   [ GWREPL_NOPROMPT=1 ] gwrepl.exe [script_arg1] ...
+*)
 (**/**) (* Utils. *)
 
+let ls_r dirs =
+  let rec loop result = function
+    | f :: fs when Sys.is_directory f ->
+      Sys.readdir f
+      |> Array.to_list
+      |> List.rev_map (Filename.concat f)
+      |> List.rev_append fs
+      |> loop (f :: result)
+    | f :: fs -> loop (f :: result) fs
+    | [] -> result
+  in
+  loop [] dirs
+in
+
 let skip_to_next_message ic =
   let rec loop () =
     let line = input_line ic in
     if Mutil.start_with "    " 0 line then line else loop ()
   in loop ()
-;;
+in
 
 let get_all_versions ic =
-  let rec loop accu =
+  let rec loop accu ref =
     let line = try input_line ic with End_of_file -> "" in
-    if line = "" then accu
+    if line = "" then (accu, ref)
     else
       try
         let i = String.index line ':' in
         let lang = String.sub line 0 i in
         let transl = String.sub line (i + 1) (String.length line - i - 1) in
-        loop ((lang, transl) :: accu)
-      with Not_found -> accu
-  in loop []
-;;
+        if lang = "->" then
```

Comment:
This is not included in geneweb. Should be removed until this feature is validated and added.